### PR TITLE
add custom model transport action, request/response to common package

### PR DESCRIPTION
**Pull request by ylwu-amzn**
Signed-off-by: Yaliang Wu <ylwu@amazon.com>

### Description
add custom model transport action, request/response to common package
Unit test will be added later. Send out PR earlier to unblock other's work.
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: common/src/main/java/org/opensearch/ml/common/transport/load/LoadModelNodeResponse.java [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+
+    public LoadModelNodeResponse(DiscoveryNode node, Map<String, String> modelLoadStatus) {
+        super(node);
+        this.modelLoadStatus = modelLoadStatus;
+    }
+    /**
+     * Constructor
+     *
+     * @param in StreamInput
+     * @throws IOException throws an IO exception if the StreamInput cannot be reML from
```
Comment (dhrubo-os) [inline common/src/main/java/org/opensearch/ml/common/transport/load/LoadModelNodeResponse.java:34]: what is reML?

File: common/src/main/java/org/opensearch/ml/common/transport/load/LoadModelNodeResponse.java [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+            this.modelLoadStatus = in.readMap(s -> s.readString(), s-> s.readString());
+        }
+
+    }
+
+    /**
+     * Creates a new UnloadNodeResponse object and reMLs in the stats from an input stream
+     *
+     * @param in StreamInput to reML from
+     * @return UnloadModelNodeResponse object corresponding to the input stream
```
Comment (dhrubo-os) [inline common/src/main/java/org/opensearch/ml/common/transport/load/LoadModelNodeResponse.java:48]: But in the code it's returning `LoadModelNodeResponse`

File: common/src/main/java/org/opensearch/ml/common/transport/upload/MLUploadInput.java [hunk 3]
```diff
@@ -0,0 +1,218 @@
...
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+/**
+ * ML input data: algirithm name, parameters and input data set.
+ */
+@Data
+public class MLUploadInput implements ToXContentObject, Writeable {
+
+    public static final String FUNCTION_NAME_FIELD = "function";
```
Comment (dhrubo-os) [inline common/src/main/java/org/opensearch/ml/common/transport/upload/MLUploadInput.java:34]: What is this function field about? is this algorithm name?

**Review by dhrubo-os — APPROVED** [review]
May be we should add an issue to refactor code later for 

1. improve test coverage
2. improve documentation
3. consistency in code styling (space, line break etc..).

**Comment by ylwu-amzn** [comment]
> May be we should add an issue to refactor code later for
> 
> 1. improve test coverage
> 2. improve documentation
> 3. consistency in code styling (space, line break etc..).

Good point. Will create issues for these.

File: common/src/main/java/org/opensearch/ml/common/transport/upload/MLUploadInput.java (hunk 3 above)
Comment (ylwu-amzn) [reply common/src/main/java/org/opensearch/ml/common/transport/upload/MLUploadInput.java:34]: Yes, this part is a little bit messy. We call it algorithm name before. Now for custom model, it seems not so reasonable to call it "algorithm". So just use a more general name `function` here. Do you have any suggestion?

File: common/src/main/java/org/opensearch/ml/common/transport/load/LoadModelNodeResponse.java (hunk 2 above)
Comment (ylwu-amzn) [reply common/src/main/java/org/opensearch/ml/common/transport/load/LoadModelNodeResponse.java:48]: Good point, will fix this later

File: common/src/main/java/org/opensearch/ml/common/transport/load/LoadModelNodeResponse.java (hunk 1 above)
Comment (ylwu-amzn) [reply common/src/main/java/org/opensearch/ml/common/transport/load/LoadModelNodeResponse.java]: Good catch, that's a wrong word. will fix

**Review by dhrubo-os — APPROVED** [review]

**Review by Zhangxunmt — APPROVED** [review]

File: common/src/main/java/org/opensearch/ml/common/transport/load/LoadModelInput.java [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+import java.io.IOException;
+
+import static org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken;
+
+@Data
+public class LoadModelInput implements ToXContentObject, Writeable {
+    private static final String MODEL_ID_FIELD = "model_id";
+    private static final String TASK_ID_FIELD = "task_id";
+    private static final String MODEL_CONTENT_HASH_FIELD = "model_content_hash";
```
[the remark this answers is not in the document]
Comment (ylwu-amzn) [reply common/src/main/java/org/opensearch/ml/common/transport/load/LoadModelInput.java:26]: Thanks, will fix this part in next PR

**Pull request merged**